Formula/erlang.rb

```diff
@@ -32,6 +33,7 @@ class Erlang < Formula
   depends_on "libtool" => :build
   depends_on "openssl"
   depends_on "fop" => :optional # enables building PDF docs
+  depends_on :java => :optional
```

Comment:
Yes, but `brew install erlang --with-javac` won't trigger the `java` dependency, because the latter doesn't grasp you meant to call it. It's an explicit match rather than word association, essentially. Probably easiest to just rename the option `with-java`, etc.